gateway/handler.go

```diff
@@ -233,6 +233,8 @@ func (i *handler) getOrHeadHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	i.handlePathAffinityHints(w, r, contentPath, logger)
```

Comment:
What about only calling this if it's a request for `raw` blocks or CAR files as indicated [here](https://github.com/ipfs/specs/pull/462/files#diff-c18ce974c093e841cec3118920c4fa12613b711446890e5149cec768c71ea0b4R102-R104)? Or do you want to extend the scope?